pkg/archive/archive.go

```diff
@@ -124,15 +125,66 @@ func IsArchivePath(path string) bool {
 	return err == nil
 }
 
+const (
+	ZstdMagicSkippableStart = 0x184D2A50
+	ZstdMagicSkippableMask  = 0xFFFFFFF0
+)
+
+var (
+	Bzip2Magic = []byte{0x42, 0x5A, 0x68}
+	GzipMagic  = []byte{0x1F, 0x8B, 0x08}
+	XzMagic    = []byte{0xFD, 0x37, 0x7A, 0x58, 0x5A, 0x00}
+	ZstdMagic  = []byte{0x28, 0xb5, 0x2f, 0xfd}
+)
+
+type matcher = func([]byte) bool
+
+func magicNumberMatcher(m []byte) matcher {
+	return func(source []byte) bool {
+		return bytes.HasPrefix(source, m)
+	}
+}
+
+// ZstDetector detects zstd compression algorithm.
+// Zstandard compressed data is made of one or more frames.
+// There are two frame formats defined by Zstandard: Zstandard frames and Skippable frames.
+// See https://tools.ietf.org/id/draft-kucherawy-dispatch-zstd-00.html#rfc.section.2 for more details.
+func ZstDetector(buf []byte) bool {
```

Comment:
Given that we don't use `ZstDetector` outside of this package; could you un-export it? (it's easy to export it if we would ever need it outside of this package, but the reverse can be more work if some external project started using it, so I prefer to be a bit conservative on exporting things 😅)

Reply:
Done.